### DISCUSSION TITLE
feat(flag): enable filtering image findings which will not be fixed

### DISCUSSION THREAD

**Discussion by jhebden-gl** [aquasecurity/trivy#5000] [Ideas]
### Description

It would be helpful to be able to filter specifically findings which will not be fixed, per advisory information and the associated fix status.

Currently, `--ignore-unfixed` is a very helpful CLI flag for returning only findings which have a fix available. However, for many uses it is also helpful to also include vulnerability findings where it is not yet known if a fix will be made available, or when it is known that a vulnerable component is definitely impacted, but a fix has not been released yet.

These additional fix statuses can be very useful for planning, & remediation SLA and reporting - as filtering these out using `--ignore-unfixed` potentially masks vulnerabilities which are actually impactful & need to be remediated in these use cases.

Adding a `--ignore-wontfix` flag which filters only on `NotAffected`, `WillNotFix` and `EndOfLife` advisory status would accomplish this goal without potentially filtering out vulnerabilities where fix availability is pending or not yet determined for a vulnerability.

### Target

Container Image

### Scanner

Vulnerability

**Comment by DmitriyLewen** [comment]
Hello @jhebden-gl 
We already added this option - https://github.com/aquasecurity/trivy/pull/4867

You can read about supported OSs here - https://aquasecurity.github.io/trivy/v0.44/docs/configuration/filtering/#by-status

Regards, Dmitriy

**Reply by jhebden-gl** [reply]
Thanks @DmitriyLewen - I should have been explicit that I was aware of `--ignore-status`. I was suggesting that it would be helpful to have a simple option to combine the common statuses which indicate a vendor would not fix a vulnerability, similar to what is configured for `--ignore-unfixed`. It was mostly a convenience item. I also only opened this as I am working on an MR for this already, so consider this less of a "feature request" but rather an "issue opened so that I can link it to an MR" :) - if you think there would be no benefit or interest in adding this convenience parameter, please feel free to leave this closed and I will not create the MR. Thanks!

**Reply by DmitriyLewen** [reply]
Hello @jhebden-gl 
We try not overload Trivy with many flags. We left out `--ignore-unfixed` because it concatenates more standard statuses and this is expected for users.
Also only RedHat and Debuan currently support non-default (i mean not fixed/affected) statuses. Your flag can be confusing.

For convenience of complex flag combinations in daily use, we have created [configuration file](https://aquasecurity.github.io/trivy/v0.44/docs/references/configuration/config-file/).
e.g. you can create `ignore-wontfix.yaml` file with only the `ignore-status` flag and use it.

**Reply by jhebden-gl** [reply]
I can see your point @DmitriyLewen about fixed vs unfixed being a clearer distinction, supported by all data sources. I will use the `--ignore-status` flag with specific configuration in my downstream work instead. Thank you!
